Observer/SalesQuoteSaveAfter.php

```diff
@@ -68,7 +78,8 @@ public function execute(Observer $observer)
         // Setting QuoteId at this point since the MaskedQuoteId is not updated when this event is dispatched,
         // MaskedQuoteId is set into the payload while the EventsTopic cron job moves rows into the Sync table
         $quote = $observer->getData('quote');
-        $klAddedToCartPayload = array_merge($klAddedToCartPayload, ['QuoteId' => $quote->getId()]);
+        $customerId = $this->checkCustomerAndReturnId($quote);
+        $klAddedToCartPayload['QuoteId'] = isset($customerId) ? "kx_identifier_$customerId" : $quote->getId();
```

Comment:
Im confused, does this make that every added to cart event for a logged in customer with always have the same QuoteId?

Reply:
No, just using double quotes to parse the customerId variable and concatenate the string

Reply:
Im not asking about the string formatting.  I am seeing that if we have a customer with ID=2 this will always become "kx_identifier_2" whether they added to cart this morning, this afternoonm and even a week from now

Reply:
Ah I see you can do it that way because it's at the end of the string... learning new things every day about php

Reply:
@klaviyojad That's right, we'll get the active quote for that specific customer.

I checked this by having multiple incognito windows and starting a session in one place, logging in into the other window and check, Magento populates the contents from the previous quote when the user is logged in so in essence, even if they are on their phone, it would essentially be the same quote if they are logged in to the same account

@cykolln you can use it anywhere, it doesn't need to be at the end of the string. It could also be used as "pronoun$customerId is a customer"